src/SFA.DAS.ApprenticeCommitments/Data/ApprenticeRepository.cs

```diff
@@ -1,28 +1,36 @@
-﻿using System;
-using System.Threading.Tasks;
+﻿using Microsoft.EntityFrameworkCore;
 using SFA.DAS.ApprenticeCommitments.Data.Models;
 using SFA.DAS.ApprenticeCommitments.Map;
 using SFA.DAS.ApprenticeCommitments.Models;
+using System.Net.Mail;
+using System.Threading.Tasks;
 
 namespace SFA.DAS.ApprenticeCommitments.Data
 {
     public class ApprenticeRepository : IApprenticeRepository
     {
-        private readonly Lazy<ApprenticeCommitmentsDbContext> _dbContext;
+        private readonly ApprenticeCommitmentsDbContext _db;
 
-        public ApprenticeRepository(Lazy<ApprenticeCommitmentsDbContext> dbContext)
+        public ApprenticeRepository(ApprenticeCommitmentsDbContext dbContext)
         {
-            _dbContext = dbContext;
+            _db = dbContext;
         }
 
         public async Task<ApprenticeModel> Add(ApprenticeModel model)
         {
             var apprentice = model.MapToApprentice();
-            var dbContext = _dbContext.Value;
-            await dbContext.AddAsync(apprentice);
-            await dbContext.SaveChangesAsync();
-
+            await _db.AddAsync(apprentice);
+            await _db.SaveChangesAsync();
```

Comment:
No the responsibility of SaveChanges belongs to the UoW decorator. We should not be doing it here

Reply:
It was already there before, I just changed the variable name ;)

I'm glad you said that though, as I wanted to look at moving the save responsibility out of the repository classes and into once place.

Reply:
yes we could rework that repo layer so it doesn't need to pass the apprenticeid around